fdbcli/GetBulkLoadStatusCommand.actor.cpp

```diff
@@ -0,0 +1,115 @@
+#include "fdbclient/IClientApi.h"
+#include "flow/Arena.h"
+#include "flow/FastRef.h"
+#include "flow/ThreadHelper.actor.h"
+
+#include "flow/actorcompiler.h" // This must be the last #include.
+
+namespace fdb_cli {
+
+ACTOR Future<Void> getBulkLoadStateByRange(Database cx, KeyRange rangeToRead) {
+	state Transaction tr(cx);
+	state Key readBegin = rangeToRead.begin;
+	state Key readEnd = rangeToRead.end;
+	state int64_t finishCount = 0;
+	state int64_t unfinishedCount = 0;
+	state RangeResult res;
+	while (readBegin < readEnd) {
+		state int retryCount = 0;
+		loop {
+			try {
+				tr.setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
+				tr.setOption(FDBTransactionOptions::LOCK_AWARE);
+				RangeResult res_ = wait(krmGetRanges(&tr,
+				                                     bulkLoadPrefix,
+				                                     KeyRangeRef(readBegin, readEnd),
+				                                     CLIENT_KNOBS->KRM_GET_RANGE_LIMIT,
+				                                     CLIENT_KNOBS->KRM_GET_RANGE_LIMIT_BYTES));
+				res = res_;
+				break;
+			} catch (Error& e) {
+				if (retryCount > 30) {
+					printf("Incomplete check\n");
+					return Void();
+				}
+				wait(tr.onError(e));
+				retryCount++;
+			}
+		}
+		for (int i = 0; i < res.size() - 1; ++i) {
+			if (res[i].value.empty()) {
+				continue;
+			}
+			const BulkLoadState bulkLoadState = decodeBulkLoadState(res[i].value);
+			if (bulkLoadState.phase == BulkLoadPhase::Complete) {
+				printf("[Complete]: %s\n", bulkLoadState.toString().c_str());
+				++finishCount;
+			} else if (bulkLoadState.phase == BulkLoadPhase::Running) {
+				printf("[Running]: %s\n", bulkLoadState.toString().c_str());
+				++unfinishedCount;
+			} else if (bulkLoadState.phase == BulkLoadPhase::Triggered) {
+				printf("[Triggered]: %s\n", bulkLoadState.toString().c_str());
+				++unfinishedCount;
+			} else if (bulkLoadState.phase == BulkLoadPhase::Invalid) {
+				printf("[NotStarted] %s\n", bulkLoadState.toString().c_str());
+				++unfinishedCount;
+			} else {
+				UNREACHABLE();
+			}
+			KeyRange range = Standalone(KeyRangeRef(res[i].key, res[i + 1].key));
+			ASSERT(range == bulkLoadState.range);
+		}
+		readBegin = res.back().key;
+	}
+
+	printf("Finished range count: %ld\n", finishCount);
+	printf("Unfinished range count: %ld\n", unfinishedCount);
+
+	return Void();
+}
+
+ACTOR Future<bool> getBulkLoadStatusCommandActor(Database cx, std::vector<StringRef> tokens) {
+	if (tokens.size() < 3) {
+		printUsage(tokens[0]);
+		return false;
+	}
+	Key rangeBegin = tokens[1];
+	Key rangeEnd = tokens[2];
+	if (rangeBegin > normalKeys.end || rangeEnd > normalKeys.end) {
+		printUsage(tokens[0]);
+		return true;
+	}
+	KeyRange range = Standalone(KeyRangeRef(rangeBegin, rangeEnd));
+	wait(getBulkLoadStateByRange(cx, range));
+
+	return true;
+}
+
+CommandFactory getBulkLoadStatusFactory("get_bulkload_status",
+                                        CommandHelp("get_bulkload_status <Begin, End>",
```

Comment:
What happens if the specified range does not match a task? If the range is > or < an existing task? Do we return task-not-found?

Reply:
Or it looks like if the range is > a single task range, we return status for each task that fits inside the range?

Reply:
The range in the command means that loading any data within the range from the input file(s). This will be checked when storage server loads the file (not implemented yet). The goal of this PR is to establish a reasonable overall structure of the bulk load so I prefer to implement the check in following PRs.

Reply:
You might update the descriptive doc. so has more clarity around how the passed ranged is treated.

Reply:
Thanks for the suggestion. I updated the doc.